File: backend/core/publication/domain/entities/publication.py
from typing import List, Literal

from backend.core.category.domain.entities.category import Category
from backend.core.common.domain.entities.entity import Entity

PublicationFormat = Literal[
    "linkedin_post",
    "blog_post",
    "short_video",
    "long_video"
]

PublicationStage = Literal[
    "preview",
    "ready"
]


class Publication(Entity):
    id: str
    title: str
    content: str
    categories: List[Category]
    outlining: List[str]
    format: PublicationFormat
    stage: PublicationStage
    thought_ids: List[str]

File: frontend/src/types/publication/publication.ts
import type { Entity } from '../common/entity';
import type { Category } from '../category/category';

export type PublicationFormat = 
  | "linkedin_post"
  | "blog_post"
  | "short_video"
  | "long_video";

export type PublicationStage = 
  | "preview"
  | "ready";

export type Publication = Entity & {
  title: string;
  content: string;
  categories: Category[];
  outlining: string[];
  format: PublicationFormat;
  stage: PublicationStage;
  thought_ids: string[];
  user_guideline: string;
}; 